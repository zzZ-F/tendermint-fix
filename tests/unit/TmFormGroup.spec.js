import { mount } from "@vue/test-utils"
import htmlBeautify from "html-beautify"
import { TmFormGroup } from "@/index"

describe("TmFormGroup", () => {
  let wrapper

  let propsData = {
    error: true,
    fieldId: "sign-in-password",
    fieldLabel: "Sign In Password"
  }

  beforeEach(() => {
    wrapper = mount(TmFormGroup, {
      propsData,
      slots: { default: '<input class="fake-field" value="fake"></div>' }
    })
  })

  it("has a field id from props", () => {
    expect(wrapper.vm.fieldId).toBe("sign-in-password")
  })

  it("has the expected html structure", () => {
    expect(htmlBeautify(wrapper.html())).toMatchSnapshot()
  })

  it("shows the label of the field", () => {
    expect(
      wrapper
        .find(".tm-form-group__label")
        .text()
        .trim()
    ).toBe("Sign In Password")
  })

  it("has a slot with a field", () => {
    expect(wrapper.findAll(".fake-field").length).toBe(1)
  })
})
