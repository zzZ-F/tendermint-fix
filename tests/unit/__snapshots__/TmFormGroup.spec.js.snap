// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TmFormGroup has the expected html structure 1`] = `
<div class="tm-form-group tm-form-group--error">
  <!---->
  <label for="sign-in-password" class="tm-form-group__label">Sign In Password</label>
  <div class="tm-form-group__field">
    <input value="fake" class="fake-field">
  </div>
</div>
`;
